clientupdate/clientupdate.go

```diff
@@ -169,6 +170,12 @@ func NewUpdater(args Arguments) (*Updater, error) {
 type updateFunction func() error
 
 func (up *Updater) getUpdateFunction() (fn updateFunction, canAutoUpdate bool) {
+	hi := hostinfo.New()
+	// We don't know how to update custom tsnet binaries, it's up to the user.
+	if hi.Package == "tsnet" {
```

Comment:
Is it possible for the application to override this value, bypassing the check? (Arguably if they do it's no worse off than they are now, so LGTM, but I was curious)

Reply:
Yep, it's populated via `hostinfo.SetPackage` just like tsnet does: https://github.com/tailscale/tailscale/blob/83de820422dbf5571781d36db31854c0dbc8c0ef/tsnet/tsnet.go#L323
And the custom tsnet apps should be using `hostinfo.SetApp` to identify themselves.